src/agent/debugger/src/stack.rs

```diff
@@ -84,15 +83,24 @@ impl Display for DebugFunctionLocation {
 #[derive(Clone, Debug, Hash, PartialEq)]
 pub enum DebugStackFrame {
     Frame {
-        function: String,
+        module_name: String,
         location: DebugFunctionLocation,
```

Comment:
This is basically what I had in mind.

The only latently weird thing is the `location` field, which has type `DebugFunctionLocation`. It is now misnamed (when `symbol` is `None`), and we are overloading it semantically.

WDYT about a struct variant like:
```rust
Frame {
    module_name: String,
    module_offset: u64,
    symbol: Option<Symbol>,
    symbol_location: Option<DebugFunctionLocation)>,
}
```
We'd maybe want to group the latter two into a single optional value, if they should always be present/absent together.

Reply:
I'm fine with that, though if we're promoting module_offset to Frame, the only entity in DebugFrameLocation is file_info, which could get promoted and renamed.  

Any thoughts on this @lzybkr?

Reply:
updated to incorporate @ranweiler's suggestions.